update `pulldown-cmark` to `0.9.3`

lukas-code opened the pull request:
This PR updates `pulldown-cmark` to version `0.9.3`, which does two main things:
* Pulls in https://github.com/raphlinus/pulldown-cmark/pull/643 to fix https://github.com/rust-lang/rust/issues/111117
* Allows parsing strikethrough with single tildes, e.g. `~foo~` -> ~foo~. This matches the [GFM spec](https://github.github.com/gfm/#strikethrough-extension-).

Full changelog: https://github.com/raphlinus/pulldown-cmark/pull/646

Commit:
rustdoc: add regression test for broken link due to double backticks

lukas-code commented:
@rustbot label -T-bootstrap -A-testsuite -S-waiting-on-review +S-blocked

Commit:
update pulldown-cmark to 0.9.3

Commit:
rustdoc: add test for strikethrough with single tildes

Also merge the two almost identical strikethrough tests together and document what the test tests.

lukas-code commented:
r? rustdoc @rustbot ready

lukas-code commented:
The strikethrough change might be worth documenting in @rustbot label relnotes

```rust
/// ~~old~~ ~new~
pub fn strike() {}
```

before:
![single strike + tildes](https://github.com/rust-lang/rust/assets/26522220/57bd60cf-0f01-4083-b02f-98297f2e6ac3)

after:
![double strike](https://github.com/rust-lang/rust/assets/26522220/e91f9b76-a495-4265-979b-66bbd43a3560)

lukas-code commented:
r? @GuillaumeGomez maybe?

lukas-code commented on tests/rustdoc/strikethrough-in-summary.rs:
```diff
@@ -1,6 +0,0 @@
-#![crate_name = "foo"]
```
I've moved this check to the other test file, because they're otherwise identical (https://github.com/rust-lang/rust/pull/111152/files#diff-d7023b89dcafe34a2d98341780db733e68362bc533a2755fbef0c959eadbddf0R3).

Should I split this test again?

GuillaumeGomez replied:
Sorry I missed that. Instead, can you write a short comment at the top of the test file explaining what it's testing please?

Commit:
rustdoc book: document single tilde strikethrough

lukas-code commented:
I've added a comment to the test and updated the rustdoc book.

I also verified that the single tilde striketrough actually renders correctly in the book.

bors commented:
:pushpin: Commit c2a446a95ec503436930dba2408a9c36953ba19f has been approved by `GuillaumeGomez`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=GuillaumeGomez c2a446a95ec503436930dba2408a9c36953ba19f -->
<!-- homu: {"type":"Approved","sha":"c2a446a95ec503436930dba2408a9c36953ba19f","approver":"GuillaumeGomez","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
